### PR TITLE
39 us as a user i want to block another user from sending me messages

### PR DESCRIPTION
**Pull request by llescure**
# Description

A user is able to block another user from the member option in the chat.
A blocked user is blurred and you can't send dm to him/her
If a dm exists between the 2 users, on block this dm channel is deleted

Block option is not available on the profile in the friends option because it created a bug in the following scenario : 
1- User 1 sends a dm message to user 2
2- User 1 is friend with user 2, user 1 goes to his profile and blocks users 2 via his friend option
3- User1 goes back to the chat, he gets a 404 error
It is due to the fact that no socket is listening on room left on the profile component. Adding it would have required to use a query to fetch all channels for the current user (lot of work and code complexity for a little feature)

# Changes include

- [X] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have added the needed labels
- [X] I have linked this PR to an issue
- [X] I have linked this PR to a milestone
- [X] I have linked this PR to a project
- [X] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: frontend/app/src/components/section-components/chat/send-dm.tsx
Comment (estelle-rcr) [inline frontend/app/src/components/section-components/chat/send-dm.tsx:26]: il faudra qu'on mette toastify ici mais on en a d'autres ;)

**Review by estelle-rcr — COMMENTED** [review]
2 general issues:
- if user 1 starts a DM with user 2, leaves the chat page (to main for instance). User 2 blocks user 1. User 1 goes back to chat: she falls on 404 and most of our 'gets' are in 404 error (the user lands on the URL of their DM which doesn't exist anymore
- if user 1 blocks user 2, user 2 blocks user 1 and then unblock user 1. User 1 appears on user 2's page as unblurred (until reload of the page as user 2 is still blocked by user 1

File: frontend/app/src/components/section-components/chat/send-dm.tsx
Comment (llescure) [reply frontend/app/src/components/section-components/chat/send-dm.tsx:26]: J'ai justement fait une issue à ce sujet ;)

File: backend/app/src/channel/channel.controller.ts
Comment (llescure) [reply backend/app/src/channel/channel.controller.ts:88]: Oui tu ne peux pas passer de body à une get

Comment (estelle-rcr) [reply backend/app/src/channel/channel.controller.ts:88]: ah yes!

**Review by cnstll — APPROVED** [review]
🍺

**Pull request merged**
